### PR TITLE
Friendly error when exec'ing to gem outside the bundle

### LINKED ISSUE #5487: Bundler Should Have Better Error Messages for Missing Deps
Upon setting up a barebones install of `brew install ruby` and setting up bundler and rake using `gem install bundler rake`, I made a simple ruby project with a Rakefile and a Gemfile. 

Anytime I'd run `bundle exec rake` it'd complain with "can't find executable rake", even though it was in the path and running correctly. As pointed out by @segiddins it's because `gem 'rake'` was missing from the Gemfile (which, to me being new to ruby, wasn't immediately obvious to be required). 

It'd be nice if there was an error message checking for this and potentially offering suggested fixes! :)

Bundler env dump and such: https://gist.github.com/b3ll/9cba83d5bcd40a9f0e4405a28cafef2e

### PR DESCRIPTION
**Pull request by segiddins**
Closes #5487

### CODE REVIEW

File: spec/commands/exec_spec.rb [hunk 1]
```diff
@@ -218,6 +218,18 @@
...
+  it "raises a helpful error when exec'ing to something outside of the bundle" do
+    install_gemfile! <<-G
+      gem "rack"
+    G
+    [true, false].each do |l|
+      bundle! "config disable_exec_load #{l}"
+      bundle "exec rake"
+      expect(err).to include "can't find executable rake for gem rake. rake is not currently included in the bundle, perhaps you meant to add it to your Gemfile?"
+    end
+    expect(out).to include "bundler: failed to load command: rake"
```
Comment (segiddins) [inline spec/commands/exec_spec.rb]: need to just remove this expectation

File: spec/commands/exec_spec.rb [hunk 2]
```diff
@@ -218,6 +218,18 @@
...
   end
 
+  it "raises a helpful error when exec'ing to something outside of the bundle" do
+    install_gemfile! <<-G
+      gem "rack"
+    G
+    [true, false].each do |l|
+      bundle! "config disable_exec_load #{l}"
+      bundle "exec rake"
+      expect(err).to include "can't find executable rake for gem rake. rake is not currently included in the bundle, perhaps you meant to add it to your Gemfile?"
```
Comment (segiddins) [inline spec/commands/exec_spec.rb]: How would you want the error message to look in the case the gem is in the bundle, but it just doesn't have the executable?

Comment (colby-swandale) [inline spec/commands/exec_spec.rb]: `Couldn't find executable "rake".` should be fine unless i'm missing something? 😧

File: spec/commands/exec_spec.rb [hunk 3]
```diff
@@ -218,6 +218,17 @@
...
```
Comment (segiddins) [inline spec/commands/exec_spec.rb]: But then it won't say which gem it was missing from?

Comment (colby-swandale) [inline spec/commands/exec_spec.rb]: my bad, keep it as it currently is, it's mainly just the last part: `included in the bundle, perhaps you meant to add it to your Gemfile?` that i suggest editing to: `included in the Gemfile`

**Comment by segiddins** [comment]
This should be good to merge, @colby-swandale feel free to submit a PR improving error message wordings?

**Comment by colby-swandale** [comment]
looks good!

**Comment by segiddins** [comment]
@bundlerbot r+

**Comment by segiddins** [comment]
@bundlerbot r+

**Comment by segiddins** [comment]
@bundlerbot r+

**Pull request merged**